src/V3Width.cpp

```diff
@@ -7309,6 +7326,63 @@ class WidthVisitor final : public VNVisitor {
         if (nodep->subDTypep()) return hasOpenArrayIterateDType(nodep->subDTypep()->skipRefp());
         return false;
     }
+    int getDepthOfClassHierarchyRec(const AstClass* const classp) {
+        // Return the number of classes between the current class and its most base class
+        // (inclusive)
+        if (!classp->extendsp())
+            return 1;
+        else
+            return 1 + getDepthOfClassHierarchyRec(classp->extendsp()->classp());
+    }
+    AstClass* getCommonBaseClass(AstClass* class1p, AstClass* class2p) {
```

Comment:
I think computeCastable does some of this - please use that and enhance if need be.